ath79: Add support for TP-Link CPE710v2

timnoack opened the pull request:
This pull request adds support for the TP-Link CPE710v2. The v2 uses the same SoC, RAM, Flash, WiFi MAC, etc., as the v1.

The differences (I spotted) to the v1 are:
* The MDIO address of the ethernet PHY changed from 0x4 to 0x0.
* The factory image updater requires the firmware version to be > 0.0.X and the compatibility level >= 1.

The pull request adding support to the tplink-safeloader in the firmware-utils repo is [here](https://github.com/openwrt/firmware-utils/pull/36).

timnoack commented:
I also got pictures of the PCB, including the locations of the UART bootlog pins (no pinheader or test points for them...). Let me know if I can upload them somewhere. Maybe on a new wiki page for the CPE720v2? I have not contributed to OpenWrt before.

timnoack commented:
Changed commit description and signed-off according to https://openwrt.org/submitting-patches

neheb commented on target/linux/ath79/dts/qca9563_tplink_cpe710-v2.dts:
```diff
@@ -0,0 +1,170 @@
+				read-only;
+			};
+
+			info: partition@60000 {
```
info is not referenced anywhere.

timnoack replied:
Removed the label

DragonBluep commented on target/linux/ath79/dts/qca9563_tplink_cpe710-v2.dts:
```diff
@@ -0,0 +1,170 @@
+		};
+
+		wlan5g {
+			label = "blue:wlan5g";
```
```suggestion
			color = <LED_COLOR_ID_BLUE>;
			function = LED_FUNCTION_WLAN_5GHZ;
```

timnoack replied:
Done

DragonBluep commented on target/linux/ath79/dts/qca9563_tplink_cpe710-v2.dts:
```diff
@@ -0,0 +1,170 @@
+
+			partition@40000 {
+				label = "u-boot-env";
+				reg = <0x040000 0x010000>;
```
If it's not a real u-boot environment partition, it can be marked as `read-only;`.

timnoack replied:
In fact, the partition reads as 0xFF only. So I guess its not a real env partition and marked it as read-only

timnoack commented:
@andyboeh I implemented a refactored version in the branch `cpe710v2_refactored` [in my fork](https://github.com/timnoack/openwrt/commits/cpe710v2_refactor/), which uses a common DTS file for v1 and v2. I am not an expert in writing DTS, but if it looks correct to you, I could switch over to it in this PR.

timnoack committed:
ath79: refactor dts of TP-Link CPE710-v1

This is done in preparation of adding support for the CPE710-v2,
which uses a similiar device tree.

Signed-off-by: Tim Noack <tim@noack.id>
Link: https://github.com/openwrt/openwrt/pull/16637
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

timnoack committed:
ath79: add support for TP-Link CPE710-v2

TP-Link CPE710-v2 is an outdoor wireless CPE for 5 GHz with one Ethernet
port based on the AP152 reference board. Compared to the CPE710-v1, the
only change observed in hardware is that the mdio address of the ethernet
physical changed from 0x4 to 0x0.

Specifications:
- SoC: QCA9563-AL3A MIPS 74kc @ 775MHz, AHB @ 258MHz
- RAM: 128MiB DDR2 @ 650MHz
- Flash: 16MiB SPI NOR Based on the GD25Q128
- Wi-Fi 5Ghz: ath10k chip (802.11ac for up to 867Mbps on 5GHz wireless
  data rate), based on the QCA9896
- Ethernet: one 1GbE port
- 23dBi high-gain directional 2×2 MIMO parabolic antenna
- Power, LAN, WLAN5G Blue LEDs

Flashing instructions:
Flash factory image through stock firmware WEB UI or through TFTP
To get to TFTP recovery just hold reset button while powering on for around
30-40 seconds and release.
Rename factory image to recovery.bin
Stock TFTP server IP:192.168.0.100
Stock device TFTP address:192.168.0.254

Signed-off-by: Tim Noack <tim@noack.id>
Link: https://github.com/openwrt/openwrt/pull/16637
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

andyboeh commented:
I'm not an expert either, I've just prepared a bunch of these. You could also move the `phy-mode` property to the common dtsi, apart from that, it looks good to me!

My device should arrive on Wednesday, so I can test the v2 support soon (at least I assume it's going to be a v2 device).

andyboeh commented:
@timnoack 

> I implemented a refactored version in the branch `cpe710v2_refactored` [in my fork](https://github.com/timnoack/openwrt/commits/cpe710v2_refactor/), which uses a common DTS file for v1 and v2. I am not an expert in writing DTS, but if it looks correct to you, I could switch over to it in this PR.

I received my units, built your branch and flashed the two devices (both v2). It worked immediately and so well, that I put them on a 950m link. They maintain a 260MBit/s link for about 24h now.

hauke commented:
> @andyboeh I implemented a refactored version in the branch `cpe710v2_refactored` [in my fork](https://github.com/timnoack/openwrt/commits/cpe710v2_refactor/), which uses a common DTS file for v1 and v2. I am not an expert in writing DTS, but if it looks correct to you, I could switch over to it in this PR.

I had a quick look and this looks good. Please replace this PR with the version from that branch.

andyboeh commented:
@timnoack any chance you can finalize this PR soon so that it will be in the upcoming 24.10 release?

andyboeh commented:
Great!
@timnoack Now that it's merged, will you create a PR for a backport to 24.10 so that we can get it into the release?

timnoack commented:
Thanks! See #17072 for the backport.
